src/app/src/util/util.js

```diff
@@ -273,6 +276,12 @@ export const getContributorFromQueryString = (qs) => {
     return parseInt(contributor, 10);
 };
 
+export const createTileURLWithQueryString = (qs, key) =>
+    `/tile/facilitygrid/${key}/{z}/{x}/{y}.pbf`.concat(isEmpty(qs) ? '' : `?${qs}`);
+
+export const createTileCacheKeyWithQSParams = (qs, key) =>
+    `${key}-${window.btoa(qs).slice(0, 8) || 'all'}`;
```

Comment:
Base64 encoding is not a good choice because "/" is one of the characters used in base64 (https://www.w3schools.com/jsref/met_win_btoa.asp). If we want to continue generating this key segment on the client we should consider:

- Requiring a hash function implementation (perhaps https://github.com/puleos/object-hash)
- Writing our own encoder.

Reply:
Oof, thanks for catching that. I pushed up 0e285a9699ba22647c828a3be9100105cbbaaa06
 which uses the object-hash library instead.